waspc/src/Wasp/Analyzer/Parser.hs

```diff
@@ -36,11 +38,24 @@ import Control.Monad.Except (runExcept)
 import Control.Monad.State (evalStateT)
 import Wasp.Analyzer.Parser.AST
 import Wasp.Analyzer.Parser.Ctx (Ctx (..), WithCtx (..), ctxFromPos, ctxFromRgn, fromWithCtx, getCtxRgn, withCtx)
-import Wasp.Analyzer.Parser.Monad (initialState)
+import Wasp.Analyzer.Parser.Monad (Parser, makeInitialState)
 import Wasp.Analyzer.Parser.ParseError
 import qualified Wasp.Analyzer.Parser.Parser as P
 import Wasp.Analyzer.Parser.SourcePosition (SourcePosition (..))
 import Wasp.Analyzer.Parser.Token
 
-parse :: String -> Either ParseError AST
-parse = runExcept . evalStateT P.parse . initialState
+isValidWaspIdentifier :: String -> Bool
+isValidWaspIdentifier str = case parseExpression str of
+  Right (Var name) -> noCharsSkipped
+    where
```

Comment:
Haha, wasn't this your idea? To document the condition?

Reply:
Yeah bat I used `let in` :D! And you used `where`. Which I didn't know is doable here! And to be honest, I like `where` better, as it reads in the right order. It is just unusual to see it here! Can you at inline the thing after where so it is `where ...` instead of `where\n...`? Then it would be perfect.

Reply:
Yes, that is why I was surprised that `where` can go here! But very cool.